Introduce SELinux policy for libvirt drivers

5umm3r15 opened the pull request:
New SELinux policy for libvirt drivers:
Hypervisor drivers:
 - virtqemud  (QEMU/KVM)
 - virtlxcd   (LXC)
 - virtvboxd  (VirtualBox)

Secondary drivers:
 - virtstoraged   (host storage mgmt)
 - virtnetworkd   (virtual network mgmt)
 - virtinterface  (network interface mgmt)
 - virtnodedevd   (physical device mgmt)
 - virtsecretd    (security credential mgmt)
 - virtnwfilterd  (ip[6]tables/ebtables mgmt)
 - virtproxyd     (proxy daemon)

SELinux policy for virtvxz and virtxend has not been created yet.

5umm3r15 committed:
Introduce SELinux policy for libvirt drivers

New SELinux policy for libvirt drivers:
Hypervisor drivers:
 - virtqemud  (QEMU/KVM)
 - virtlxcd   (LXC)
 - virtvboxd  (VirtualBox)

Secondary drivers:
 - virtstoraged   (host storage mgmt)
 - virtnetworkd   (virtual network mgmt)
 - virtinterface  (network interface mgmt)
 - virtnodedevd   (physical device mgmt)
 - virtsecretd    (security credential mgmt)
 - virtnwfilterd  (ip[6]tables/ebtables mgmt)
 - virtproxyd     (proxy daemon)

SELinux policy for virtvxz and virtxend has not been created yet.

5umm3r15 commented:
Hi @wrabcak @zpytela 
can you please review it?
It is a design of selinux policy for virt drivers.
SELinux policy for virtvxz and virtxend has not been created yet. 
Interfaces are still not solved yet.

wrabcak commented on virt.te:
```diff
@@ -304,6 +309,62 @@ files_type(virt_qemu_ga_data_t)
+virt_driver_template(virtnetworkd)
+
+type virtnetworkd_var_run_t;
+files_pid_file(virtnetworkd_var_run_t)
```
You can put declaration of *_var_run_t type to the `virt_driver_template()` same as you did with process domains.

5umm3r15 replied:
ok, I will do it

wrabcak commented on virt.te:
```diff
@@ -304,6 +309,62 @@ files_type(virt_qemu_ga_data_t)
+#virtxend
+type virtxend_t;
+type virtxend_exec_t;
+init_daemon_domain(virtxend_t, virtxend_exec_t)
```
Why virt_driver_template is not used here? Is it for purpose?

5umm3r15 replied:
I skipped policy for virtxend driver for now.

wrabcak commented on virt.te:
```diff
@@ -1674,6 +1771,213 @@ corenet_rw_tun_tap_dev(virt_bridgehelper_t)
+allow virtnwfilterd_t self:rawip_socket create_socket_perms;
+
+allow virtnwfilterd_t virt_var_run_t:dir { create search_dir_perms };
+manage_dirs_pattern(virtnwfilterd_t, virtnwfilterd_var_run_t, virtnwfilterd_var_run_t)
```
These set of rules could be part of `virt_driver_template`

5umm3r15 replied:
ok, thanks

5umm3r15 committed:
Split virt policy, introduce virt_supplementary module

Separate the services from the original virt files that are not libvirt related and create virt_supplementary policy module.

5umm3r15 committed:
fixup! Introduce SELinux policy for libvirt drivers

5umm3r15 committed:
fixup! fixup! Introduce SELinux policy for libvirt drivers

5umm3r15 committed:
fixup! fixup! fixup! Introduce SELinux policy for libvirt drivers
